### PR TITLE
--no-logging and --no-coverage should only affect XML configuration

### LINKED ISSUE #4401: Add CLI flag to generate only one type of CC & Logging regardless of phpunit.xml
Hi, I'm the current maintainer of https://github.com/paratestphp/paratest, a tool to run PHPUnit in parallel, leveraging sub-processes.

A feature we must provide is to generate code-coverage and logging accordingly to `phpunit.xml` specifications.

Our tool should:

1. Read `phpunit.xml` `<filter/coverage>` and `<logging>` sections, which may all set to all the available configurations
1. Disable all of them during ParaTest process
1. Enable **only** `coverage-php` and `log-junit` into PHPUnit sub-processes
1. Wait for sub-processes to finish and to generate the only two cc+log above
1. Merge everything together within ParaTest process
1. Generate the desired cc/logs respecting again `phpunit.xml` `<filter/coverage>` and `<logging>` sections

As of yet this can't be done because if `<filter/coverage>` and `<logging>` sections are specified in `phpunit.xml`, cc and logs are always generated except with `--no-coverage`/`--no-logging` active, but with `--no-coverage`/`--no-logging` active no cc nor logs are generated at all.

The workaround we are using right now is suggesting *not* to have any section in `phpunit.xml` except the `<whitelist>` one, and specify each option in the CLI, so ParaTest can easily rewrite them as wish, but of course this is sub-optimal.

So I'm asking something like `vendor/bin/phpunit --no-coverage --coverage-php tmp.php --no-logging --log-junit tmp.log`.

If you so kind to  write down few instructions to how and where implement such feature, I'm happy to propose a PR.

### PR DESCRIPTION
**Pull request by Slamdunk**
Close #4401

### CODE REVIEW

File: src/TextUI/TestRunner.php [hunk 1]
```diff
@@ -341,77 +341,71 @@ public function run(TestSuite $suite, array $arguments = [], array $warnings = [
         $coverageFilterFromOption            = false;
         $codeCoverageReports                 = 0;
 
-        if (!isset($arguments['noLogging'])) {
```
Comment (Slamdunk) [inline src/TextUI/TestRunner.php:344]: It's worth viewing TestRunner changes with whitespace changes hidden:
https://github.com/sebastianbergmann/phpunit/pull/4446/files?w=1

File: tests/end-to-end/no-log-cc-override/no-log-no-cc.phpt [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+phpunit -c _files/phpunit.xml --no-logging --log-junit php://stdout _files/NoLogNoCcTest.php
+--FILE--
+<?php declare(strict_types=1);
+$arguments = [
+    '--configuration',
+    \realpath(__DIR__ . '/_files/phpunit.xml'),
+    '--no-logging',
+    '--log-junit',
+    'php://stdout',
+//    '--no-coverage',
```
Comment (Slamdunk) [inline tests/end-to-end/no-log-cc-override/no-log-no-cc.phpt:11]: At start I redacted the test also with `--no-coverage` coverage, but was unable to load a driver within the PHPT test.

Any guess on how to test this properly?

**Comment by Slamdunk** [comment]
Is there anything I can do to have this PR released within `9.5`?

**Comment by sebastianbergmann** [comment]
This was merged manually.

**Pull request closed without merging**